proto/viam/app/v1/app.proto

```diff
@@ -1155,6 +1156,12 @@ message UpdateRegistryItemRequest {
   string description = 3;
   Visibility visibility = 4;
   optional string url = 5;
+
+  oneof metadata {
+    ModuleUpdateMetadata module_update_metadata = 6;
+    MLModelUpdateMetadata ml_model_update_metadata = 7;
+    MLTrainingUpdateMetadata ml_training_update_metadata = 8;
```

Comment:
why are these called __UpdateMetadata?

Reply:
Explained in this [scope ammendment](https://docs.google.com/document/d/1XPXGwMfxzMkt7tFJeU6ibg1fBh7vl9UvuQokiq6Oiq8/edit?tab=t.0#bookmark=id.o8k9d7813hlq)

```
I made a mistake in the scope and used the same proto messages for Update and Get APIs. Some fields are not allowed to be updated so I have created different messages to separate get/update
```

Reply:
ah makes sense, so these are the editable fields per registry item. [opt, nit] would `UpdateModuleMetadata` or `ModuleMetadataUpdate` be clearer?

Reply:
Yeah I can change to `UpdateModuleMetadata`. Makes sense